lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/service/CriCheckingService.java

```diff
@@ -269,6 +272,11 @@ && requiresAuthoritativeSourceCheck(newVcs, sessionVcs)) {
         return JOURNEY_NEXT;
     }
 
+    private void setFailedIdentityCheckOnIpvSessionItem(IpvSessionItem ipvSessionItem) {
+        ipvSessionItem.setFailureCode(ReverificationFailureCode.IDENTITY_CHECK_FAILED);
+        ipvSessionItem.setFailureDescription("Identity check failed.");
+    }
```

Comment:
An important gotcha here: updating the session object will not automatically persist the change* - you need to call `ipvSessionService.updateIpvSession(ipvSessionItem)` to update it in DynamoDB.

Ideally we want to minimise the number of times we do this, but it's better to make sure it happens - even if some later step throws an exception! (We already update the session earlier in `ProcessCriCallbackHandler`, for example).

*It does when running locally, but this is a flaw in our local setup - see https://github.com/govuk-one-login/ipv-core-back/pull/2139/files/f936f6e3fd0edce5a5d092059612907688e53378#r1668974060 for some discussion